Allow use of GitHub Access Token via GITHUB_TOKEN

Lewiscowles1986 opened the pull request:
Fixes #3

Lewiscowles1986 committed:
Allow use of GitHub Access Token via GITHUB_TOKEN

Fixes #3

PurpleBooth commented on change-github-default-branch.sh:
```diff
@@ -2,17 +2,22 @@
-if [ $# -ne 3 ] && [ $# -ne 2 ]; then
-	echo "USAGE: $0 pat_token user/repo [new_default]" 1>&2
+if [ $# -ne 3 ] && [ $# -ne 2 ] && [ -z "$GITHUB_TOKEN" ] && [ $# -ne 1 ]; then
+	echo "USAGE: $0 [pat_token] user/repo [new_default]" 1>&2
```
I don't like that the first argument is optional, can we make it second or last?

(oh why oh why didn't I use getopts 😄)

Lewiscowles1986 replied:
I can. There is also a logical error in this if people do supply their token. I'm working on a fix

Lewiscowles1986 committed:
move token to be second argument

Address changes requested in #4 as well as logical inconsistency where token ENV is set, but different TOKEN is provided as CLI argument

Lewiscowles1986 committed:
check greater than or equal to, not doesn't equal

PurpleBooth commented:
Amazing thanks!
